File: RohiniM_Assessment24_Pytest/Assessment24/Pages/Register_Page.py
class register_page():
    myaccount_menu = "(//li[@class='dropdown']/child::a)[1]"
    register_menu = "//ul[@class='dropdown-menu dropdown-menu-right']/child::li[1]/child::a"
    
    first_name = "//input[@id='input-firstname']"
    last_name = "//input[@id='input-lastname']"
    email = "//input[@id='input-email']"
    telephone = "//input[@id='input-telephone']"
    password = "//input[@id='input-password']"
    confirm_password = "//input[@id='input-confirm']"
    privacy_policy = "//input[@name='agree']"
    submit = "//input[@value='Continue']"
    success_msg = "//h1[text()='Your Account Has Been Created!']"
    error_msg = "//div[@class='alert alert-danger alert-dismissible']"

File: RohiniM_Assessment25_Playwright/Assessment/Pages/RegisterPage.ts
import {Page, Locator, expect} from '@playwright/test';

export class RegisterPage{
    readonly page: Page;
    readonly registerPageTitle: Locator;
    readonly firstName: Locator;
    readonly lastName: Locator;
    readonly email: Locator;
    readonly telephone: Locator;
    readonly password: Locator;
    readonly confirmPassword: Locator;
    readonly privacyPolicy: Locator;
    readonly continueButton: Locator;
    readonly accountCreated: Locator;

    constructor(page: Page){
        this.page = page;
        this.registerPageTitle = page.getByRole('heading', {name: 'Register Account'});
        this.firstName = page.locator("input#input-firstname");
        this.lastName = page.locator("input#input-lastname");
        this.email = page.getByPlaceholder("E-Mail");
        this.telephone = page.getByPlaceholder("Telephone");
        this.password = page.locator("input#input-password");
        this.confirmPassword = page.locator("input#input-confirm");
        this.privacyPolicy = page.locator("//input[@name='agree']");
        this.continueButton = page.locator("//input[@type='submit']");
        this.accountCreated = page.getByText('Your Account Has Been Created!')
    }

    async verifyRegisterPage(){
        await expect(this.registerPageTitle).toBeVisible();
    }

    async enterPersonalDetails(f_name: string, l_name: string, e_mail: string, tele: string,){
        await this.firstName.fill(f_name);
        await this.lastName.fill(l_name);
        await this.email.fill(e_mail);
        await this.telephone.fill(tele);
    }

    async enterPasswordDetails(pass: string, confirm_pass: string){
        await this.password.fill(pass);
        await this.confirmPassword.fill(confirm_pass);
    }

    async agreePolicy(){
        await this.privacyPolicy.check();
    }

    async clickContinueButton(){
        await this.continueButton.click()
    }

    async verifyAccountCreation(){
        await expect(this.accountCreated).toBeVisible({timeout: 50000});
    }
}